spec/lib/tasks/email_notes_to_admins_rake_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+
+# see http://robots.thoughtbot.com/test-rake-tasks-like-a-boss
+describe 'email_notes_to_admins' do
+  include_context 'rake'
+
+  before(:all) { FactoryGirl.create(:app_config) }
+
+  before(:each) do
+    @notes = FactoryGirl.create(:checked_out_reservation,
+                                notes: 'here is a note')
+    @no_notes = FactoryGirl.create(:checked_out_reservation)
+  end
+
+  it 'sends emails for checked out reservations that have unsent notes' do
+    @notes.update_attributes(notes_unsent: true)
+    expect(@notes.notes_unsent).to be_truthy
+    @notes.save!
+    expect { subject.invoke }.to(
+      change { ActionMailer::Base.deliveries.count }.by(1))
+  end
+
+  it "doesn't send emails for checked out reservations with sent notes" do
+    expect(@notes.notes_unsent).to be_falsey
+    expect { subject.invoke }.to_not(
+      change { ActionMailer::Base.deliveries.count })
+  end
+
+  it 'sends emails for checked in reservations that have unsent notes' do
+    @notes = FactoryGirl.build(:checked_in_reservation,
```

Comment:
~~Didn't you already create this object [above](https://github.com/YaleSTC/reservations/pull/1293/files#diff-a2997a8dba03ec31c3ffc45ae8269528R10)? Why don't you create all three reservations in the `before_each` block - `@checked_out_notes`, `@checked_in_notes`, and `@no_notes`.~~ There is a *ton* of duplication here... I think we can simplify a bit. How about something along these lines:

1. Create an array of two symbols: `[:checked_out_reservation, :checked_in_reservation]`
2. Create a shared example that takes a symbol as a parameter to feed into the two FactoryGirl calls (`@notes` and `@no_notes`). This example could include both checks (if the flag is true or false). You could potentially also feed the symbol into the title of the `it` block.
3. Then, loop over the symbol array and call the shared example for each case.

For an example of something like this see [here](https://github.com/YaleSTC/reservations/blob/master/spec/features/rails_admin_spec.rb#L4) (obviously written for feature specs but the structure should be very similar). Let me know if that makes sense!